src/content_script/mastodonInject.js

```diff
@@ -121,28 +118,47 @@ async function injectInteractionButtons() {
     });
 }
 
+/**
+ * Initialise injection for feedElement.
+ * 
+ * @returns {void}
+ */
+async function injectFeed() {
+    const observer = new MutationObserver(() => {
+        injectInteractionButtons();
+    });
+
+    try {
+        const feedElement = await waitForElement(
+            "#mastodon .item-list[role='feed']",
+            false,
+        );
+
+        observer.observe(feedElement, {
+            childList: true, subtree: true,
+        });
+    } catch (error){
+        // feedElement not found
+    }
+}
+
 /**
  * Initialise injection for all remote Mastodon buttons.
  * 
  * @returns {void}
  */
-async function init() {
+function init() {
     injectFollowButton();
 
-    const observer = new MutationObserver(() => {
-        Promise.allSettled([
-            injectInteractionButtons(),
-        ]);
-    });
+    const ogType = document.querySelector("meta[property='og:type']");
 
-    const feedElement = await waitForElement(
-        "#mastodon .item-list[role='feed']",
```

Comment:
@rugk should this next version of the Simplified Federation Extension support Mastodon v3?

If so, will this change reduce that support, do you guess? 

[ I ask, and did not research myself, since you do have more experience. If you don't know .. I can dig around in v3 source ( v3.5.5 still being worked on https://github.com/mastodon/mastodon/commit/696f7b360882e71ff45b6ddafc8eea17184a0f31 ) and could then make an educated guess. ]

Reply:
Hi @nrichards nope AFAIK this will not affect support for v3 because the CSS classes and the type of HTML elements used in v3 are different, so the CSS selectors for the followButton and the interactionButtons will not match.

Reply:
Awesome! This makes sense to me .. thanks for taking the time to consider and respond to my concern.

Reply:
Ah thanks, but good point. Even though old versions may not be _that_ many I can imagine some forks or so could keep the old implementation or so, so we need to stay aware of it.
Note this is not only theory, this happened in practice before already, see https://github.com/rugk/mastodon-simplified-federation/issues/71#issuecomment-1153266408.

So thanks a lot!